cylc/flow/network/resolvers.py

```diff
@@ -216,20 +219,17 @@ def node_filter(node, node_type, args):
     else:
         # live objects can be represented by a universal ID
         tokens = Tokens(node.id)
-    state = getattr(node, 'state', None)
     return (
         (
-            args.get('ghosts')
-            or state
-            or node_type in DEF_TYPES
-        )
-        and (
-            not args.get('states')
+            (
+                state is None
+                or not args.get('states')
```

Comment:
So if the node has no state (how does this happen?) then we include it in filter results irrespective of whether a query state was specified? Or am I misunderstanding?

Reply:
> So if the node has no state (how does this happen?)
It can happen if it's a task definition or something like that.
> then we include it in filter results irrespective of whether a query state was specified?
Correct, as we use the same filter for nodes without states (task defs, family defs)